src/Core/Repositories/EntityFramework/MaintenanceRepository.cs

```diff
@@ -0,0 +1,43 @@
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using AutoMapper;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Bit.Core.Repositories.EntityFramework
+{
+    public class MaintenanceRepository : BaseEntityFrameworkRepository, IMaintenanceRepository
+    {
+        public MaintenanceRepository(IServiceScopeFactory serviceScopeFactory, IMapper mapper)
+            : base(serviceScopeFactory, mapper)
+        { }
+
+        public async Task DeleteExpiredGrantsAsync()
+        {
+            using (var scope = ServiceScopeFactory.CreateScope())
+            {
+                var dbContext = GetDatabaseContext(scope);
+                var query = from g in dbContext.Grants
+                            where g.ExpirationDate < DateTime.UtcNow
+                            select g;
+                dbContext.RemoveRange(query);
+                await dbContext.SaveChangesAsync();
+            }
+        }
+
+        public Task DisableCipherAutoStatsAsync()
+        {
+            throw new NotImplementedException();
```

Comment:
Wondering if these should `throw` or just silently do nothing, `return Task.Completed`;

Reply:
This is a good point. I threw an exception initially because they in fact are not implemented methods, but I'm honestly not sure what that will break down the chain. Changed to return a completed task.